### PR TITLE
feat(kalert): redesign [khcp-2008]

### PR DESCRIPTION
**Pull request by aanchalm01**
### Summary
KAlert was redesigned to allow for different dismiss types(none/icon/button), alert type(banner/alert), different size(small/large). 

#### Changes made:
* Add new props: type, dismissType, size(large)
* Add new icon/buttons
* Style icons/buttons based on appearance
* Update test snapshots

Screen Shot 2021-11-01 at 1 16 23 PM
Screen Shot 2021-11-01 at 1 16 51 PM
Screen Shot 2021-11-01 at 1 25 18 PM

### PR Checklist
- [x] Does not introduce dependencies
- [x] **Functional:** all changes do not break existing APIs and if so, bump major version.
- [x] **Tests pass:** check the output of yarn test packages/<Kongponent>
- [x] **Naming:** the files and the method and prop variables use the same naming conventions as other Kongponents
- [x] **Framework style:** abides by the essential rules in Vue's style guide
- [x] **Cleanliness:** does not have formatting issues, unused code (e.g., console.logs), or leftover comments
- [x] **Docs:** includes a technically accurate README, uses JSDOC where appropriate

### CODE REVIEW

File: packages/KAlert/KAlert.vue [hunk 1]
```diff
@@ -13,31 +15,71 @@
...
-        size="12"
-        view-box="0 0 12 12" />
+        size="14" />
     </button>
+    <div
+      :class="appearance"
+      class="k-alert-action ml-3">
+      <!-- @slot Use this slot to pass extra buttons other than Dismiss  -->
+      <slot
+        v-if="hasActionButtons && (dismissType === 'button' || dismissType ==='none')"
```
Comment (kaiarrowood) [inline packages/KAlert/KAlert.vue:37]:
```suggestion
        v-if="hasActionButtons"
```

Let's just not constrain it on dismissType.

File: packages/KAlert/KAlert.vue [hunk 2]
```diff
@@ -13,31 +15,71 @@
...
+    <span v-if="size === 'large'">
+      <KIcon
+        :size="iconSize"
+        :color="iconColor"
+        :icon="icon ? icon : 'notificationInbox'"
+        class="k-alert-icon" />
     </span>
+    <div class="k-alert-msg-text">
+      <span
+        :class="type === 'banner' && (size === 'large') ? 'k-alert-text' : ''"
```
Comment (kaiarrowood) [inline packages/KAlert/KAlert.vue:64]:
```suggestion
        :class="type === 'banner' && size === 'large' ? 'k-alert-text' : ''"
```

File: docs/components/alert.md [hunk 3]
```diff
@@ -44,20 +50,254 @@ What color and purpose the Alert should be. Shares similar appearances to those
...
+
+- `isShowing`
+
+<KAlert
+  alert-message="isShowing: true by default"/>
+
+```vue
+<KAlert
+  alert-message="isShowing: true by default"/>
+```
```
Comment (kaiarrowood) [inline docs/components/alert.md:292]: No need for lines 286 -> 292

File: docs/components/alert.md [hunk 4]
```diff
@@ -44,20 +50,254 @@ What color and purpose the Alert should be. Shares similar appearances to those
...
+<KAlert
+  :is-showing=false
+  alert-message="isShowing set to false"/>
+
 ```vue
 <KAlert
-  class="dismissible"
-  is-dismissible
-  alert-message="I can be dismissed!"/>
+  alert-message="isShowing set to false"/>
```
Comment (kaiarrowood) [inline docs/components/alert.md:300]:
```suggestion
  :is-showing="false"
  alert-message="isShowing set to false"/>
```

File: docs/components/alert.md [hunk 5]
```diff
@@ -44,20 +50,254 @@ What color and purpose the Alert should be. Shares similar appearances to those
...
+  alert-message="isShowing: true by default"/>
+
+```vue
+<KAlert
+  alert-message="isShowing: true by default"/>
+```
+
+<KAlert
+  :is-showing=false
+  alert-message="isShowing set to false"/>
```
Comment (kaiarrowood) [inline docs/components/alert.md:296]: We can remove lines 294 - 296

**Review by kaiarrowood — CHANGES_REQUESTED** [review]
Looks like Dismiss buttons don't dismiss when clicked:
image

Let's use this approach whenever we want to show all four styles (ie. all four examples, one single code block). For sections, `Props` -> `type` -> `banner` and `Props` -> `type` -> `alert`, and `Props` -> `dismissType`.

image

File: packages/KAlert/KAlert.vue (hunk 1 above)
Comment (aanchalm01) [reply packages/KAlert/KAlert.vue]: Fixed!

File: packages/KAlert/KAlert.vue (hunk 2 above)
Comment (aanchalm01) [reply packages/KAlert/KAlert.vue]: Fixed!

File: docs/components/alert.md (hunk 3 above)
Comment (aanchalm01) [reply docs/components/alert.md]: Removed!

File: docs/components/alert.md (hunk 4 above)
Comment (aanchalm01) [reply docs/components/alert.md:278]: Fixed!

File: docs/components/alert.md (hunk 5 above)
Comment (aanchalm01) [reply docs/components/alert.md]: Removed!

**Review by kaiarrowood — APPROVED** [review]
Woohoo!!! KAlert COMPLETE!! 🐉 🔥 🔥 🔥

**Pull request merged**